sdk/purview/azure-analytics-purview-account/src/main/java/com/azure/analytics/purview/account/PurviewAccountClientBuilder.java

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.analytics.purview.account;
+
+import com.azure.analytics.purview.account.implementation.PurviewAccountClientImpl;
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.serializer.JacksonAdapter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** A builder for creating a new instance of the PurviewAccountClient type. */
+@ServiceClientBuilder(
+        serviceClients = {
+            AccountsClient.class,
+            CollectionsClient.class,
+            ResourceSetRulesClient.class,
+            AccountsAsyncClient.class,
+            CollectionsAsyncClient.class,
+            ResourceSetRulesAsyncClient.class
+        })
+public final class PurviewAccountClientBuilder {
+    private static final String SDK_NAME = "name";
+
+    private static final String SDK_VERSION = "version";
+
+    static final String[] DEFAULT_SCOPES = new String[] {"https://purview.azure.net/.default"};
```

Comment:
It is visited in PurviewAccountClientTestBase.java
```
policies.add(new BearerTokenAuthenticationPolicy(credential, PurviewAccountClientBuilder.DEFAULT_SCOPES));
```
This `TestBase` class is copied from exisiting `TestBase` in other SDKs. It can be private. `TestBase` does not have to create a new HTTP pipeline. It can add its own custom policies in client builder via `addPolicy` method.
https://github.com/Azure/azure-sdk-for-java/pull/23666/files/99ae2d53da9c65fd7699d447f05985e8b156571f#diff-119213e7caf442222fe7d175987dbb480754a89640a2751515064c31041eb2c6R46

Reply:
It can be - but it's non-trivial - let's visit this later in a separate issue.

Reply:
Created an [issue](https://github.com/Azure/autorest.java/issues/1167)